mohamad-dakdouk/week5/wed/D23-countries-cities-hm/app/controllers/countries_controller.rb

```diff
@@ -0,0 +1,62 @@
+class CountriesController < ApplicationController
+
+def home
+
+end
+
+
+#create
+def new
+    @countries = Country.new
+
+end
+
+def create
+
+    Country.create(
+            name: params[:country][:name],
+            location: params[:country][:location],
+            language: params[:country][:language],
+            population: params[:country][:population],
+            vist: params[:country][:vist],
+            image_url: params[:country][:image_url]
+    )
+    redirect_to countries_index_path
+end
+
+#read
+def index
+    @countries = Country.all
+end
+
+def show
+    @countries = Country.find params[:id]
+end
+
+#update
+def edit
+    @countries = Country.find params[:id]
+
+end
+
+def update
+    country = Country.find params[:id]
+  country.update(
+            name: params[:country][:name],
+            location: params[:country][:location],
+            language: params[:country][:language],
+            population: params[:country][:population],
+            vist: params[:country][:vist],
+            image_url: params[:country][:image_url]
+    )
+    redirect_to country_path(country.id)
+
+end 
+
+def destroy
+    Country.destroy params[:id]
+    redirect_to country_path
```

Comment:
once you have destroyed it, you're redirecting it back to the path it was just at. This causes an error cause the country no longer exists. 

try redirect it back to the index page instead